### PR TITLE
DH rename test file, remove compute tag tests, make pypp test more thorough

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

- Rename Test_DampedHarmonicGaugeQuantities to Test_DampedHarmonic
- Remove the damped harmonic compute tags test. These are tested in the initialization action test and will be moved into the DH initialization action in a future PR
- Make the pypp tests thorough enough to justify deleting the C++ tests.
  The only thing that isn't tested are the integers and that's because pypp currently doesn't play nice with non-double types.

Only the following commits are new:
- Rename Test_DampedHarmonicGaugeQuantities to Test_DH
- Remove damped harmonic compute tags test
- Make DH pypp test more thorough

Depends on:
- [x] #2293 
- [x] #2296 

### Types of changes:

- [ ] Bugfix
- [x] New feature
- [x] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by nilsdeppe** [comment]
@geoffrey4444 @moxcodes I've rebased this on develop and should be ready for review

File: src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.hpp [hunk 1]
```diff
@@ -120,17 +117,17 @@ namespace gauges {
...
+ *   +& A_{L1} \mathrm{log}(\sqrt{g}/N)^{e_{L1} + 1} \partial_a [R(t)
  * W(x^i)],\\
  *
  * \partial_a \mu_2
- *  =& \partial_a [A_{L2} R_{L2}(t) W(x^i) \mathrm{log}(1/N)^{e_{L2} + 1}], \\
- *  =& A_{L2} R_{L2}(t) W(x^i) \partial_a [\mathrm{log}(1/N)^{e_{L2} + 1}] \\
- *     +& A_{L2} \mathrm{log}(1/N)^{e_{L2} + 1} \partial_a [R_{L2}(t) W(x^i)],
+ *  =& \partial_a [A_{L2} R(t) W(x^i) \mathrm{log}(1/N)^{e_{L2} + 1}], \\
+ *  =& A_{L2} R(t) W(x^i) \partial_a [\mathrm{log}(1/N)^{e_{L2} + 1}] \\
+ *     +& A_{L2} \mathrm{log}(1/N)^{e_{L2} + 1} \partial_a [R(t) W(x^i)],
```
Comment (geoffrey4444) [inline src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.hpp:130]: So I'm guessing SpEC had all these options, but in practice we always set them all to the same time scales, so why keep all those options here? Makes sense, but might be worth commenting in the documentation that we allowed different time scales in the past (e.g. cite the relevant reference, maybe your paper?) but in practice we use a single time scale. Up to you

**Review by geoffrey4444 — APPROVED** [review]
Looks good, but one suggested optional change to the docs

**Review by moxcodes — APPROVED** [review]
Looks good

**Review by geoffrey4444 — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**